corehq/apps/app_manager/static/app_manager/js/hq.app_data_drilldown.ng.js

```diff
@@ -0,0 +1,248 @@
+                    $scope.isLoaded = true;
+                })
+                .error(self._handleError);
+        };
+        self._initializeForm();
+
+        $scope.updateAppChoices = function () {
+            var app_choices = self._apps_by_type[$scope.formData.app_type];
+            util.setApps(app_choices);
+            $scope.selectedAppData = {};
+            $scope.selectedFormData = {};
+            $scope.hasNoCaseTypes = false;
+            util.setModules();
+            util.setForms();
+            util.setCaseTypes();
+        };
+
+        $scope.updateModuleChoices = function () {
+            var module_choices = self._modules_by_app[$scope.formData.application];
+            util.setModules(module_choices);
+            $scope.selectedFormData = {};
+            util.setForms();
+        };
+
+        $scope.updateFormChoices = function () {
+            if ($scope.formData.application && $scope.formData.module) {
+                var form_choices = self._forms_by_app_by_module[$scope.formData.application][$scope.formData.module];
+                util.setForms(form_choices);
+            }
+        };
+
+        $scope.updateCaseTypeChoices = function () {
+            var case_type_choices = self._case_types_by_app[$scope.formData.application];
+            util.setCaseTypes(case_type_choices);
+            $scope.hasNoCaseTypes = _.isEmpty(case_type_choices);
+        };
+
+        $scope.handleSubmitForm = function () {
+            $scope.isSubmittingForm = true;
+
+            // Immediately copy form data object so that if modal is
+            // accidentally dismissed, data is still properly sent
+            var formData = _.clone($scope.formData);
+
+            djangoRMI[djangoRMICallbackName]({
+                formData: formData
+            })
+            .success(function (data) {
+                if (data.success) {
+                    processApplicationDataFormSuccessCallback(data);
+                } else {
+                    $scope.isSubmittingForm = false;
+                    $scope.formSubmissionError = data.error;
+                }
+            })
+            .error(function () {
+                $scope.isSubmittingForm = false;
+                $scope.formSubmissionError = 'default';
+            });
+        };
+
+        $scope.selectedAppData = {};
+        $scope.$watch('formData.application', function (curApp) {
+            var currentApps = _.filter(self._apps_by_type[$scope.formData.app_type], function (app) {
+                return app.id === curApp;
+            });
+            if (currentApps.length === 0) {
+                $scope.selectedAppData = {};
+            } else {
+                var app_data = currentApps[0].data;
+                app_data.name = currentApps[0].text;
+                $scope.selectedAppData = app_data;
+            }
+        });
+
+        $scope.selectedFormData = {};
+        $scope.$watch('formData.form', function (curForm) {
```

Comment:
hey, yeah, let's chat today to make sure nothing gets missed, I'll add that in a separate PR after I address the more minor stuff here. it's behind a feature flag so prod won't be affected yet